src/Core/Addon/AddonInfo.php

```diff
@@ -14,32 +14,102 @@
  */
 final class AddonInfo
 {
+	/**
+	 * Parse addon comment in search of addon infos.
+	 *
+	 * like
+	 * \code
+	 *   * Name: addon
+	 *   * Description: An addon which plugs in
+	 * . * Version: 1.2.3
+	 *   * Author: John <profile url>
+	 *   * Author: Jane <email>
+	 *   * Maintainer: Jess without link
+	 *   * Maintainer: Robin <email>
+	 *   * Status: in development
+	 * \endcode
+	 *
+	 * @internal Never create this object by yourself, use `Friendica\Core\Addon\AddonHelper::getAddonInfo()` instead.
+	 * @see Friendica\Core\Addon\AddonHelper::getAddonInfo()
+	 *
+	 * @param string $addonId the name of the addon
+	 * @param string $raw The raw file content
+	 */
+	public static function fromString(string $addonId, string $raw): self
+	{
+		$data = [
+			'id' => $addonId,
+		];
+
+		$result = preg_match("|/\*.*\*/|msU", $raw, $m);
+
+		if ($result === false || $result === 0) {
+			return self::fromArray($data);
+		}
+
+		$ll = explode("\n", $m[0]);
+
+		foreach ($ll as $l) {
+			$l = trim($l, "\t\n\r */");
+			if ($l !== '') {
```

Comment:
```suggestion
			if (!empty($l)) {
```
Empty is null, false, string with zero length

Reply:
`$l` could only be a string. Why should we use a loose check instead?